pyne/alara.py

```diff
@@ -890,5 +892,242 @@ def _find_phsrc_dc(idc, phtn_src_dc):
+    matlib_file = os.path.join(run_dir, "matlib")
+    _gt_write_matlib(mats, matlib_file)
+
+    # Write fluxin file
+    fluxin_file = os.path.join(run_dir, "fluxin")
+    fluxes = []
+    for m in range(num_mats):
+        for n in range(num_n_groups):
+            fluxes.append([neutron_spectrum[n] if x ==
+                           n else 0 for x in range(num_n_groups)])
+        fluxes.append(neutron_spectrum) # total spectrum
+        fluxes.append([0]*175) # blank spectrum
+    _gt_write_fluxin(fluxes, fluxin_file)
+
+    # Write geom file
+    input_file = os.path.join(run_dir, "inp")
+    phtn_src_file = os.path.join(run_dir, "phtn_src")
+    # Two extra zones are needed; one for the whole spectrum
+    # and one for the zero spectrum. Total number of zones = num_n_group + 2
+    num_n_groups += 2
+    _gt_write_inp(run_dir, data_dir, mats, num_n_groups, flux_magnitudes, 
+                  irr_times, decay_times, input_file, matlib_file,
+                  fluxin_file, phtn_src_file, num_p_groups)
+
+    # Run ALARA
+    sub = subprocess.Popen(['alara',input_file], stderr=subprocess.STDOUT,
+                           stdout=subprocess.PIPE).communicate()[0]
+    return phtn_src_file
+
+def calc_eta(data_dir, mats, neutron_spectrum, flux_magnitudes, irr_times,
+             decay_times, num_p_groups, run_dir, clean):
+    """
+    Function that returns eta values (SNILB check result) for each material 
+    and each decay time
+    
+    Parameters
+    ----------
+    data_dir : str
+        Path to directory containing nuclib file
+    mats : list of Material objects
+        List of properties of materials in the geometry
+    neutron_spectrum : list
+        Neutron spectrum (length is equal to number of n energy groups)
+    flux_magnitudes : list
+        Magnitude of flux in each neutron energy group
+    irr_times : list
+        Irradiation times [s]
+    decay_times : list
+        Decay times [s]
+    num_p_groups : int
+        The number of photon energy groups for ALARA calculation
+    run_dir: str
+        Directory to store ALARA run input and output files    
+    clean : bool
+        If True, remove run_dir
+        
+    Returns
+    ----------
+    eta : numpy.ndarray
+        eta value for each material listed.  This is a 2D array
+        [mat, decay_time]
+    """
+    num_n_groups = len(neutron_spectrum)
+    num_mats = len(mats)
+    num_decay_times = len(decay_times)
+    
+    # Run ALARA
+    if not os.path.exists(run_dir):
+        os.makedirs(run_dir)    
+    phtn_src_file = _gt_alara(data_dir, mats, neutron_spectrum, flux_magnitudes, 
+                              irr_times, decay_times, num_p_groups, run_dir)
+    # Parse ALARA output
+    sup = np.zeros(shape=(num_mats, num_decay_times))
+    tot = np.zeros(shape=(num_mats, num_decay_times))
+    zero = np.zeros(shape=(num_mats, num_decay_times))
+    with open(phtn_src_file, 'r') as f:
+        i = 0
+        for line in f.readlines():
```

Comment:
This block is very difficult to follow.  Better variable names and/or more comments would be helpful